pravega/flink-connectors
# [issue-216] Update expired Pravega certs and ensure system test works

Signed-off-by: Vijay Srinivasaraghavan <vijayaraghavan.srinivasaraghavan@emc.com>

**Change log description**
  * removed old certs and replaced it with Pravega standalone certs
  * fixed failing test cases specific to TLS integration
  * Updated Pravega to latest snapshot version

**Purpose of the change**
To address https://github.com/pravega/flink-connectors/issues/216

**What the code does**
- Updated Pravega dependency to latest snapshot version. 
- Replaced old certs for Standalone cluster with the latest from Pravega. 
- Fixed TLS specific tests that were failing due to missing truststore certs file. 
- Tested system test cases for both `InProc` as well as `External` Pravega cluster

**How to verify it**
- ensure build succeeds `./gradlew clean build`
- to test with external/satndalone Pravega follow below steps

From the Pravega checked out repo directory, make the following changes to the `conf/standalone-config.properties`
```
config.properties` file
singlenode.enableAuth=true
singlenode.userName=admin
singlenode.passwd=1111_aaaa
singlenode.passwdFile=../config/passwd
singlenode.enableTls=true
singlenode.keyFile=../config/key.pem
singlenode.certFile=../config/cert.pem
singlenode.keyStoreJKS=../config/standalone.keystore.jks
singlenode.keyStoreJKSPasswordFile=../config/standalone.keystore.jks.passwd
singlenode.trustStoreJKS=../config/standalone.truststore.jks
```
To ensure the server's certificate is trusted, import the certs it into the JVM's truststore. Followed below steps  (in Linux) with the provided certificate file "cert.pem":

   * `cd /path/to/pravega/config`
   * Convert the 'cert.pem' file to DER format: `openssl x509 -in cert.pem -inform pem -out cert.der -outform der`
   * Import the certificate into the local JVM's trust store:
   `sudo keytool -importcert -alias local-CA -keystore /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts  -file cert.der` (using the default password 'changeit')

Run Pravega standalone mode using command ``./gradlew startStandalone``. 

From the connector checked out repo, run the following command
`./gradlew clean test -Dpravega.uri=tls://localhost:9090`
